### PR TITLE
[FLINK-2815] [REFACTOR] Remove Pact from class and file names since it is no longer valid reference

### PR DESCRIPTION
**Pull request by hsaputra**
Remove Pact word from class and file names in Apache Flink.
Pact was the name used in Stratosphere time to refer to concept of distributed datasets (similar to Flink Dataset). It was used when Pact and Nephele still separate concept.

As part of 0.10.0 release cleanup effort, let's remove the Pact names to avoid confusion.

The PR also contains small cleanups (sorry):
1. Small refactor DataSinkTask and DataSourceTask to follow Java7 generic convention creation new collection. Remove LOG.isDebugEnabled check.
2. Simple cleanup to update MapValue and TypeInformation with Java7 generic convention creation new collection.
3. Combine several exceptions that have same catch operation.

Apologize for the extra changes with PR. But I separated them into different commits for easier review.

### CODE REVIEW

**Comment by hsaputra** [comment]
Since this change involve renaming classes I will merge this end of day tomorrow if no more comments.

File: flink-tests/src/test/resources/logback-test.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
         <appender-ref ref="STDOUT"/>
     </root>
 
-    <!--<logger name="org.apache.flink.runtime.operators.RegularPactTask" level="OFF"/>-->
+    <!--<logger name="org.apache.flink.runtime.operators.RegularTaskvel="OFF"/>-->
```
Comment (chiwanpark) [inline flink-tests/src/test/resources/logback-test.xml]: Although this line is commented, but there is a typo: `level="OFF"`.

File: flink-staging/flink-tez/src/main/java/org/apache/flink/tez/runtime/TezTask.java [hunk 2]
```diff
@@ -402,7 +402,7 @@ private void initInputLocalStrategy(int inputNum) throws Exception {
 						localStub = initStub(userCodeFunctionType);
 					} catch (Exception e) {
 						throw new RuntimeException("Initializing the user code and the configuration failed" +
-								e.getMessage() == null ? "." : ": " + e.getMessage(), e);
+								(e.getMessage() == null ? "." : ": ") + e.getMessage(), e);
```
Comment (chiwanpark) [inline flink-staging/flink-tez/src/main/java/org/apache/flink/tez/runtime/TezTask.java]: It seems wrong parenthesis. If `e.getMessage()` is `null`, there is concatenation of `"."` and `null`.

**Comment by chiwanpark** [comment]
Looks good to merge except minor issues for me. :+1:

File: flink-staging/flink-tez/src/main/java/org/apache/flink/tez/runtime/TezTask.java (hunk 2 above)
Comment (hsaputra) [inline flink-staging/flink-tez/src/main/java/org/apache/flink/tez/runtime/TezTask.java]: You are right, mea culpa from me.

File: flink-tests/src/test/resources/logback-test.xml (hunk 1 above)
Comment (hsaputra) [inline flink-tests/src/test/resources/logback-test.xml]: Ah, good catch, will make the change.

**Comment by hsaputra** [comment]
Hi @chiwanpark , thanks for the review. Thanks for catching the typo and miss parentheses.
Just push the update.

**Comment by StephanEwen** [comment]
If you are cleaning up anyways, how about renaming the `RegularTask` to `BatchTask`? That is what it is, after all, the core task class for batch operations. There is also a `StreamTask`, so it would mirror that naming...

**Comment by hsaputra** [comment]
Makes sense, will make that update

**Comment by hsaputra** [comment]
Updated based on review.

**Comment by fhueske** [comment]
LGTM

**Comment by hsaputra** [comment]
Thanks all for the reviews. Merging ...

**Pull request closed without merging**